### PR TITLE
runtime: removing envoy.reloadable_features.allow_upstream_filters

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: Medium (lots of code sites)
Testing: n/a
Docs Changes: n/a
Release Notes: inline
Fixes https://github.com/envoyproxy/envoy/issues/26898

### CODE REVIEW

File: source/common/router/upstream_request.cc [hunk 1]
```diff
@@ -290,11 +274,6 @@ void UpstreamRequest::decodeHeaders(Http::ResponseHeaderMapPtr&& headers, bool e
   }
   stream_info_.response_code_ = static_cast<uint32_t>(response_code);
 
-  if (paused_for_connect_ && response_code == 200) {
```
Comment (adisuissa) [inline source/common/router/upstream_request.cc:293]: I'm surprised this wasn't guarded by `allow_upstream_filters_` as well.

File: source/common/router/upstream_request.cc [hunk 2]
```diff
@@ -750,82 +651,10 @@ void UpstreamRequest::onPoolReady(std::unique_ptr<GenericUpstream>&& upstream,
 
   for (auto* callback : upstream_callbacks_) {
     callback->onUpstreamConnectionEstablished();
-    return;
-  }
-
-  auto* headers = parent_.downstreamHeaders();
```
Comment (adisuissa) [inline source/common/router/upstream_request.cc:756]: I'm probably missing something, but seems that the removed code from this point won't be called if there are no upstream-callbacks. I guess that if the tests pass, then this case is handled correctly.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
PR should probably be cross-company reviewed. @mattklein123 can you PTAL?

File: source/common/router/upstream_request.cc (hunk 1 above)
Comment (alyssawilk) [reply source/common/router/upstream_request.cc:293]: it was implicitly - we set paused_for_connect false when we get response headers from the codec in that path.  that and encodeBodyAndTrailers is a no-op in the new mode anyway

File: source/common/router/upstream_request.cc (hunk 2 above)
Comment (alyssawilk) [reply source/common/router/upstream_request.cc:756]: it won't be called if there were upstream callbacks - the return I removed above took care of that.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**